packages/hint-manifest-icons/src/hint.ts

```diff
@@ -0,0 +1,220 @@
+/**
+ * @fileoverview Checks usage of Web App Manifest icons
+ */
+import { URL } from 'url';
+import * as getImageData from 'image-size';
+import imageType, { ImageTypeResult } from 'image-type';
+import { IHint, NetworkData, HintContext } from 'hint';
+import { ManifestEvents, ManifestParsed, ManifestImageResource } from '@hint/parser-manifest';
+import { debug as d } from '@hint/utils';
+
+import meta from './meta';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class ManifestIconHint implements IHint {
+    public static readonly meta = meta;
+    public constructor(context: HintContext<ManifestEvents>) {
+        const getIconFullPath = (iconSrc: string, hostnameWithProtocol: string) => {
+            return `${hostnameWithProtocol}/${iconSrc}`;
+        };
+
+        /**
+         * See if the `icon` file actually
+         * exists and is accessible.
+         */
+        const iconExists = async (iconPath: string) => {
```

Comment:
While I generally prefer omitting types that can be inferred, return types in function signatures are one exception because they can easily become something you didn't expect without reading the full implementation.
```suggestion
        const iconExists = async (iconPath: string): Buffer | null => {
```

Reply:
Added `Promise<Buffer | null>` as it returns a promise.